src/Vehicle/Vehicle.cxx

```diff
@@ -0,0 +1,210 @@
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+//======================================================================================
+//
+// Copyright (c) 2008-2020 the Urho3D project.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#include <Urho3D/Core/Context.h>
+#include <Urho3D/Graphics/DebugRenderer.h>
+#include <Urho3D/Graphics/DecalSet.h>
+#include <Urho3D/Graphics/Material.h>
+#include <Urho3D/Graphics/Model.h>
+#include <Urho3D/Graphics/ParticleEffect.h>
+#include <Urho3D/Graphics/ParticleEmitter.h>
+#include <Urho3D/Graphics/StaticModel.h>
+#include <Urho3D/IO/Log.h>
+#include <Urho3D/Physics/CollisionShape.h>
+#include <Urho3D/Physics/Constraint.h>
+#include <Urho3D/Physics/PhysicsEvents.h>
+#include <Urho3D/Physics/PhysicsWorld.h>
+#include <Urho3D/Physics/RaycastVehicle.h>
+#include <Urho3D/Resource/ResourceCache.h>
+#include <Urho3D/Scene/Scene.h>
+#include "Vehicle.hxx"
+
+using namespace Urho3D;
+
+const float CHASSIS_WIDTH = 2.6f;
+
+void Vehicle::RegisterObject(Context* context) {
+    context->RegisterFactory<Vehicle>();
+    URHO3D_ATTRIBUTE("Controls Yaw", float, controls_.yaw_, 0.0f, AM_DEFAULT);
+    URHO3D_ATTRIBUTE("Controls Pitch", float, controls_.pitch_, 0.0f, AM_DEFAULT);
+}
+
+Vehicle::Vehicle(Urho3D::Context* context) : LogicComponent(context) {
+    SetUpdateEventMask(USE_FIXEDUPDATE | USE_POSTUPDATE);
+    engineForce_ = 0.0f; // NOT CORRECT VALUES
+    brakingForce_ = 50.0f;
+    maxEngineForce_ = 2500.0f;
+    wheelRadius_ = 0.5f;
+    suspensionRestLength_ = 0.6f;
+    wheelWidth_ = 0.4f;
+    suspensionStiffness_ = 14.0f;
+    suspensionDamping_ = 2.0f;
+    suspensionCompression_ = 4.0f;
+    wheelFriction_ = 1000.0f;
+    rollInfluence_ = 0.12f;
```

Comment:
Probably want these in the header file as default values

Reply:
I am not sure if I want to do that, could you please explain why?

Reply:
If you only set a value in the constructor it might as well be default initialized in the header, it saves space and is cleaner.
Ill turn that question around, why would you want to keep these in the constructor?

Reply:
I think it is easier to access if they are in .cxx file and/or maybe because I still have java mindset.

Reply:
Even in Java you would default initialize in the header (aka the class definition).
here a small example:
What you currently do basically:
```cpp
class test {
    int x;
    test() {
        x = 1;
        ....
    }
}
```
What you should do:
```cpp
class test {
    int x = 1;
    test() { ... }
}
```

Accessibility doesn't get hurt in the slightest.

Reply:
make sense, I will change them.